### PR TITLE
Add doc for migration with binlog event filter and pt ghost

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
add doc for migration with binlog event filter and pt ghost

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/7480
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: data-migration/migrate-with-pt-ghost.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+title: Continuous Replication from Databases that Use gh-ost or pt-osc
+summary: Learn how to use DM to replicate incremental data from databases that use online DDL tools gh-ost or pt-osc
+---
+
+# Continuous Replication from Databases that Use gh-ost or pt-osc
+
+In production scenarios, table locking during DDL execution can block the reads from or writes to the database to a certain extent. Therefore, online DDL tools are often used to execute DDLs to minimize the impact on reads and writes. Common DDL tools are [gh-ost](https://github.com/github/gh-ost) and [pt-osc](https://www.percona.com/doc/percona-toolkit/3.0/pt-online-schema-change.html).
+
+After migrating data from MySQL to TiDB by using DM, you can enbale `online-ddl` to allow collaboration of DM and gh-ost or pt-osc.
```
Comment (lance6716) [inline data-migration/migrate-with-pt-ghost.md:10]:
```suggestion
When using DM to migrate data from MySQL to TiDB, you can enbale `online-ddl` to allow collaboration of DM and gh-ost or pt-osc.
```

not after

File: data-migration/migrate-with-pt-ghost.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+The workflow of DM:
+
+- Skip creating the ghost table downstream.
+
+- Record DDLs applied to the ghost table.
+
+- Replicate data only from the ghost table.
+
+- Apply DDLs downstream.
```
Comment (lance6716) [inline data-migration/migrate-with-pt-ghost.md:55]:
```suggestion
- Apply DDLs recorded downstream.
```

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

File: data-migration/migrate-with-binlog-event-filter.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  rule-1:
+    schema-pattern: "test_*"
+    table-pattern: "t_*"
+    events: ["truncate table", "drop table"]
+    sql-pattern: ["^DROP\\s+PROCEDURE", "^CREATE\\s+PROCEDURE"]
+    action: Ignore
+```
+
+- `schema-pattern`/`table-pattern`: Filter matching schemas or tables
+- `events`: Filter binlog events. Supported events are listed in the table below:
```
Comment (hfxsd) [inline data-migration/migrate-with-binlog-event-filter.md:30]:
```suggestion
- `schema-pattern`/`table-pattern`: Filters matching schemas or tables
- `events`: Filters binlog events. Supported events are listed in the table below:
```

File: data-migration/migrate-with-binlog-event-filter.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  | drop database   | DDL  | Drop database event   |
+  | create table    | DDL  | Create table event    |
+  | create index    | DDL  | Create index event    |
+  | drop table      | DDL  | Drop table event      |
+  | truncate table  | DDL  | Truncate table event  |
+  | rename table    | DDL  | Rename table event    |
+  | drop index      | DDL  | Drop index event      |
+  | alter table     | DDL  | Alter table event     |
+
+- `sql-pattern`：Filter specified DDL SQL statements. The matching rule supports using a regular expression.
```
Comment (hfxsd) [inline data-migration/migrate-with-binlog-event-filter.md:53]:
```suggestion
- `sql-pattern`：Filters specified DDL SQL statements. The matching rule supports using a regular expression.
```

File: data-migration/migrate-with-binlog-event-filter.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    table-pattern: "t_*"
+    events: ["truncate table", "drop table", "delete"]
+    action: Ignore
+  filter-schema-rule:
+    schema-pattern: "test_*"
+    events: ["drop database"]
+    action: Ignore
+```
+
+### Migrate sharding DML operations only
```
Comment (hfxsd) [inline data-migration/migrate-with-binlog-event-filter.md:89]:
```suggestion
### Migrate only DML operations of sharded schemas and tables
```

**Review by hfxsd — APPROVED** [review]
Rest LGTM

File: data-migration/migrate-with-pt-ghost.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+For the detailed replication instruction, refer to the following documents by scenarios:
+
+- [Migrate MySQL Data of Less Than 1 TiB to TiDB](/data-migration/migrate-mysql-tidb-less-tb.md)
+- [Migrate MySQL Data of More Than 1 TiB to TiDB](/data-migration/migrate-mysql-tidb-above-tb.md)
+- [Migrate and Merge MySQL Shards of Less Than 1 TiB to TiDB](/data-migration/migrate-shared-mysql-tidb-less-tb.md)
+- [Migrate and Merge MySQL Shards of More Than 1 TiB to TiDB](/data-migration/migrate-shared-mysql-tidb-above-tb.md)
+
+## Enable online-ddl on DM
+
+When configuring the task configuration file of DM, set the global parameter `online-ddl` to `true`, as shown below:
```
Comment (shichun-0415) [inline data-migration/migrate-with-pt-ghost.md:21]:
```suggestion
In the task configuration file of DM, set the global parameter `online-ddl` to `true`, as shown below:
```

File: data-migration/migrate-with-pt-ghost.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+- Apply DDLs recorded downstream.
+
+![dm-online-ddl](/media/dm/dm-online-ddl.png)
+
+The change in the workflow brings the following advantages:
+
+- The downstream TiDB does not need to create and replicate the ghost table, saving the storage space and network transmission overhead.
+
+- When you migrate and merge data from sharded tables, the RENAME operation is ignored for each sharded ghost table to ensure the correctness of the replication.
+
```
Comment (lichunzhu) [inline data-migration/migrate-with-pt-ghost.md:64]: Some documents are missing:
```
如果您想深入了解其实现原理，请阅读以下两篇技术博客：

- [DM 源码阅读系列文章（八）Online Schema Change 迁移支持](https://pingcap.com/blog-cn/dm-source-code-reading-8/#dm-源码阅读系列文章八online-schema-change-迁移支持)
- [TiDB Online Schema Change 原理](https://pingcap.com/zh/blog/tidb-source-code-reading-17)
```

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

File: data-migration/migrate-with-pt-ghost.md (hunk 7 above)
Comment (shichun-0415) [reply data-migration/migrate-with-pt-ghost.md:64]: > Some documents are missing:
> 
> ```
> 如果您想深入了解其实现原理，请阅读以下两篇技术博客：
> 
> - [DM 源码阅读系列文章（八）Online Schema Change 迁移支持](https://pingcap.com/blog-cn/dm-source-code-reading-8/#dm-源码阅读系列文章八online-schema-change-迁移支持)
> - [TiDB Online Schema Change 原理](https://pingcap.com/zh/blog/tidb-source-code-reading-17)
> ```

Sharp eyes!
We have no English version of these blogs and I have confirmed with Zhaoyang about it. He suggested we keep the links in Chinese but remove them from the English version. Do you have any other suggestions?

File: data-migration/migrate-with-pt-ghost.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+In production scenarios, table locking during DDL execution can block the reads from or writes to the database to a certain extent. Therefore, online DDL tools are often used to execute DDLs to minimize the impact on reads and writes. Common DDL tools are [gh-ost](https://github.com/github/gh-ost) and [pt-osc](https://www.percona.com/doc/percona-toolkit/3.0/pt-online-schema-change.html).
+
+When using DM to migrate data from MySQL to TiDB, you can enbale `online-ddl` to allow collaboration of DM and gh-ost or pt-osc.
+
+For the detailed replication instruction, refer to the following documents by scenarios:
+
+- [Migrate MySQL Data of Less Than 1 TiB to TiDB](/data-migration/migrate-mysql-tidb-less-tb.md)
+- [Migrate MySQL Data of More Than 1 TiB to TiDB](/data-migration/migrate-mysql-tidb-above-tb.md)
+- [Migrate and Merge MySQL Shards of Less Than 1 TiB to TiDB](/data-migration/migrate-shared-mysql-tidb-less-tb.md)
+- [Migrate and Merge MySQL Shards of More Than 1 TiB to TiDB](/data-migration/migrate-shared-mysql-tidb-above-tb.md)
```
Comment (shichun-0415) [inline data-migration/migrate-with-pt-ghost.md:17]:
```suggestion
- [Migrate and Merge MySQL Shards of Less Than 1 TiB to TiDB](/data-migration/migrate-sharding-mysql-tidb-less-tb.md)
- [Migrate and Merge MySQL Shards of More Than 1 TiB to TiDB](/data-migration/migrate-sharding-mysql-tidb-above-tb.md)
```

File: data-migration/migrate-with-binlog-event-filter.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+---
+
+# Filter Binlog Events
+
+This document describes how to filter binlog events when you use DM to perform continuous incremental data replication. For the detailed replication instructions, refer to the following documents by scenarios:
+
+- [Migrate MySQL Data of Less Than 1 TiB to TiDB](/data-migration/migrate-mysql-tidb-less-tb.md)
+- [Migrate MySQL Data of More Than 1 TiB to TiDB](/data-migration/migrate-mysql-tidb-above-tb.md)
+- [Migrate and Merge MySQL Shards of Less Than 1 TiB to TiDB](/data-migration/migrate-shared-mysql-tidb-less-tb.md)
+- [Migrate and Merge MySQL Shards of More Than 1 TiB to TiDB](/data-migration/migrate-shared-mysql-tidb-above-tb.md)
```
Comment (shichun-0415) [inline data-migration/migrate-with-binlog-event-filter.md:13]:
```suggestion
- [Migrate and Merge MySQL Shards of Less Than 1 TiB to TiDB](/data-migration/migrate-sharding-mysql-tidb-less-tb.md)
- [Migrate and Merge MySQL Shards of More Than 1 TiB to TiDB](/data-migration/migrate-sharding-mysql-tidb-above-tb.md)
```

File: data-migration/migrate-with-pt-ghost.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+In the task configuration file of DM, set the global parameter `online-ddl` to `true`, as shown below:
+
+```yaml
+# ----------- Global configuration -----------
+## ********* Basic configuration *********
+name: test                      # The name of the task. Should be globally unique.
+task-mode: all                  # The task mode. Can be set to `full`, `incremental`, or `all`.
+shard-mode: "pessimistic"       # The shard merge mode. Optional modes are `pessimistic` and `optimistic`. The `pessimistic` mode is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set it to the "optimistic" mode.
+meta-schema: "dm_meta"          # The downstream database that stores the `meta` information.
+online-ddl: true                # Enable online-ddl support on DM to support automatic processing of "gh-ost" and "pt" for the upstream database.
```
Comment (shichun-0415) [inline data-migration/migrate-with-pt-ghost.md:30]:
```suggestion
online-ddl: true                # Enable online-ddl support on DM to support automatic processing of "gh-ost" and "pt-osc" for the upstream database.
```

File: data-migration/migrate-with-pt-ghost.md (hunk 7 above)
Comment (lichunzhu) [reply data-migration/migrate-with-pt-ghost.md:64]: > He suggested we keep the links in Chinese but remove them from the English version.
SGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**